src/psyclone/configuration.py

```diff
@@ -748,27 +753,86 @@ class DynConfig(APISpecificConfig):
-                "section of the config file: {0}".format(str(err)),
-                config=self._config)
+        # Ref. to parent Config object
+        self._config = config
+        # Initialise redundant computation setting
+        self._compute_annexed_dofs = None
+        # Initialise LFRic datatypes' default kinds (precisions) settings
+        self._default_kind = {}
+        # Set mandatory keys
+        # TODO: to be fully populated here for LFRic (Dynamo0.3) API in #282
+        # when Dynamo0.1 API is removed.
+        self._mandatory_keys = []
+
+        # TODO: This "if" clause will become redundant in #282 as there will be
+        # just LFRic (Dynamo0.3) configuration
+        if section.name == "dynamo0.3":
+            # Define and check mandatory keys
+            self._mandatory_keys = ["access_mapping",
+                                    "COMPUTE_ANNEXED_DOFS", "default_kind"]
+            mdkeys = {key.lower() for key in self._mandatory_keys}
+            if not mdkeys.issubset(set(section.keys())):
+                raise ConfigurationError(
+                    "Missing mandatory configuration option in the "
+                    "'[dynamo0.3]' section of the configuration file '{0}'. "
+                    "Valid options are: '{1}'."
+                    .format(config.filename, self._mandatory_keys))
+
+            # Parse setting for redundant computation over annexed dofs
+            try:
+                self._compute_annexed_dofs = section.getboolean(
+                    "COMPUTE_ANNEXED_DOFS")
+            except ValueError as err:
+                raise ConfigurationError(
+                    "error while parsing COMPUTE_ANNEXED_DOFS in the "
+                    "[dynamo0.3] section of the config file: {0}"
+                    .format(str(err)), config=self._config)
+
+            # Parse setting for default kinds (precisions)
+            all_kinds = self.create_dict_from_string(section["default_kind"])
+            # Set default kinds (precisions) from config file
+            from psyclone.dynamo0p3 import SUPPORTED_FORTRAN_DATATYPES
+            # Check for valid datatypes (filter to remove empty values)
+            datatypes = set(filter(None, all_kinds.keys()))
+            if datatypes != set(SUPPORTED_FORTRAN_DATATYPES):
+                raise ConfigurationError(
+                    "Invalid datatype found in the '[dynamo0.3]' "
+                    "section of the configuration file '{0}'. "
+                    "Valid datatypes are: '{1}'."
+                    .format(config.filename, SUPPORTED_FORTRAN_DATATYPES))
+            # Check for valid kinds (filter to remove any empty values)
+            datakinds = set(filter(None, all_kinds.values()))
+            if len(datakinds) != len(set(SUPPORTED_FORTRAN_DATATYPES)):
+                raise ConfigurationError(
+                    "Did not find default kind for one or more "
+                    "datatypes in the '[dynamo0.3]' section "
```

Comment:
Please include the list of `datakinds` and `SUPPORTED_FORTRAN_DATATYPES` in the error message.

Reply:
Done (I hope I understood the comment correctly). I had to use `sorted(datakinds)` in the error message to fix the set order.